app/code/Magento/Checkout/view/frontend/web/js/action/select-payment-method.js

```diff
@@ -7,11 +7,40 @@
  * @api
  */
 define([
-    '../model/quote'
-], function (quote) {
+    'mage/storage',
+    './get-totals',
+    '../model/quote',
+    '../model/full-screen-loader',
+    '../model/url-builder',
+    'Magento_Customer/js/model/customer'
+], function (storage, getTotalsAction, quote, fullScreenLoader, urlBuilder, customer) {
     'use strict';
 
     return function (paymentMethod) {
+        var serviceUrl, payload;
+
         quote.paymentMethod(paymentMethod);
+
+        delete paymentMethod.title;
+
+        payload = {
+            method: paymentMethod
+        };
+
+        if (customer.isLoggedIn()) {
+            serviceUrl = urlBuilder.createUrl('/carts/mine/selected-payment-method', {});
+        } else {
+            serviceUrl = urlBuilder.createUrl('/guest-carts/:cartId/selected-payment-method', {
+                cartId: quote.getQuoteId()
+            });
+        }
+        fullScreenLoader.startLoader();
+        storage.put(
+            serviceUrl,
+            JSON.stringify(payload)
+        ).success(function () {
```

Comment:
I believe the call to selected-payment-method can fail. In such cases we would have an endless fullScreenLoader.

Reply:
I understand, Will fix it.